Classes/Wwwision/Neos/MailChimp/Domain/Service/MailChimpService.php

```diff
@@ -159,6 +159,9 @@ public function subscribe(string $listId, string $emailAddress, array $additiona
         if ($marketingPermissions !== null) {
             $arguments['marketing_permissions'] = $marketingPermissions;
         }
+        if ($interestGroups !== null) {
+            $arguments['interests'] = array_map(function() { return true; }, array_flip($interestGroups));
```

Comment:
Does it make sense to _expect_ the `$interestGroups` array already in that format such that we can explicitly _remove_ groups, too.
i.E. ```php
$service->subscribe($listId, $emailAddress, null, null, ['group1' => true, 'group2' => false]);
```

or how does the API expect _changed_ to the interest groups?

Reply:
Thanks for checking. I'd say, lets make this an `array<string, bool>` as expected by the API. The Form Finisher just needs to make sure to send all keys then

Reply:
The Service is changed.

But the Form Finisher can't ensure that, as not all interest groups must get into it. E.g if you have 3 checkboxes with interest group ids, just the checked once will be sent to it.

Reply:
```html
<input type="hidden" name="group[foo]" value="0" />
<input type="checkbox" name="group[foo]" value="1" />

<input type="hidden" name="group[bar]" value="0" />
<input type="checkbox" name="group[bar]" value="1" />
```
should work though

Reply:
I see, but does this also work with Neos Form (grouping, same identfier for same field)?

Reply:
I would suggest, to keep it as it is. So it would work for most cases OOTB. And if someone needs to subscribe/unsubscribe it's easy to adopt in a custom finisher.

Reply:
Yes, I agree!
Thanks again for your work and patience, I will merge this one as is